barretenberg/cpp/src/barretenberg/eccvm/eccvm_prover.cpp

```diff
@@ -179,12 +182,18 @@ void ECCVMProver::execute_transcript_consistency_univariate_opening_round()
         batching_scalar *= ipa_batching_challenge;
     }
 
-    // TODO(https://github.com/AztecProtocol/barretenberg/issues/922): We are doing another round of IPA here with
-    // exactly the same labels and no domain separation so if/when labels are going to matter we are clashing.
+    std::array<OpeningClaim, 2> opening_claims = { multivariate_to_univariate_opening_claim,
+                                                   { .polynomial = batched_univariate,
+                                                     .opening_pair = { evaluation_challenge_x, batched_evaluation } } };
+
+    // Reduce the opening claims to a single opening claim via Shplonk
+    const OpeningClaim batched_opening_claim = Shplonk::prove(commitment_key, opening_claims, transcript);
+
+    // Compute the opening proof for the batched opening claim with the univariate PCS
     PCS::compute_opening_proof(
-        commitment_key, { evaluation_challenge_x, batched_evaluation }, batched_univariate, transcript);
+        commitment_key, batched_opening_claim.opening_pair, batched_opening_claim.polynomial, transcript);
```

Comment:
I won't ask you to do this because it might be annoying/spiral but it would be nice to make this method operate on `ProverOpeningClaim` like you did with some of the others.. up to you

Reply:
dont think it will spiral so will do